### PR TITLE
feature/newchanges-added-to-csvfiles

### PR DESCRIPTION
**Pull request by Pallavi0717**

### CODE REVIEW

File: phase1/pallavi/database.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    return create_engine(f'mysql+mysqlconnector://{user}:{password}@{host}/{database}')
+
+def create_table(engine, table_name, columns):
+    metadata = MetaData()
+    table = Table(table_name, metadata, *columns)
+    metadata.create_all(engine)
+    print(f"Table '{table_name}' created or already exists.")
+    return table
+
+#database has no change
```
Comment (lawbringr1) [inline phase1/pallavi/database.py:13]: no comments like these

File: phase1/pallavi/file_processing.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from pathlib import Path
+import pandas as pd
+from sqlalchemy import Column, Integer, String, Float, BigInteger, DateTime, TEXT
+from sqlalchemy.exc import SQLAlchemyError
+import logging
+
+def list_files(directory_path):
+    return [file for file in directory_path.iterdir() if file.is_file()]
+
+def check_directory_structure(directory_path):
```
Comment (lawbringr1) [inline phase1/pallavi/file_processing.py:11]: what is the use of this?

File: phase1/pallavi/file_processing.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        elif pd.api.types.is_datetime64_any_dtype(col_data) and not col_data.isna().all():
+            columns.append(Column(col_name, DateTime))
+        else:
+            columns.append(Column(col_name, String(255)))
+    return columns
+
+def load_file_to_db(engine, file_path, table_name):
+    chunk_size = 15000
+    i = 0
+    for chunk in pd.read_csv(file_path, chunksize=chunk_size):
```
Comment (lawbringr1) [inline phase1/pallavi/file_processing.py:37]: use enumurate

File: phase1/pallavi/main.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    current_path = Path.cwd()
+    logging.info(f"Current working directory: {current_path}")
+    directory_path = current_path / 'venv/modlixdata1'
+    if not directory_path.exists() or not directory_path.is_dir():
+        logging.error(f"Directory '{directory_path}' does not exist or is not a directory.")
+        sys.exit(1)
+    
+    if not check_directory_structure(directory_path):
+        logging.error(f"Directory '{directory_path}' does not have the proper structure.")
+        sys.exit(1)
```
Comment (lawbringr1) [inline phase1/pallavi/main.py:24]: we throw error not sys.exit(1)

File: phase1/pallavi/file_processing.py (hunk 2 above)
Comment (Pallavi0717) [reply phase1/pallavi/file_processing.py:11]: To check which types of files are present eg: data1.csv  ,  data2.json etc

File: phase1/pallavi/main.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import logging
+from urllib.parse import quote
+import pandas as pd
+from pathlib import Path
+from database import get_engine, create_table
+from file_processing import list_files, infer_table_schema, load_file_to_db, check_directory_structure
+logging.basicConfig(level=logging.INFO)
+
+def main():
+    try:
```
Comment (lawbringr1) [inline phase1/pallavi/main.py:10]: dont add this much code in exception. throw generic exception

File: phase1/pallavi/main.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        user = '****'
+        password = quote('********')
+        host = '**********'
+        cwd = Path.cwd()
+        logging.info(f"Current working directory: {cwd}")
+        directory_path = cwd / 'venv' / 'data'
```
Comment (lawbringr1) [inline phase1/pallavi/main.py:16]: dont save in venv -> that is for python enviorement

Comment (lawbringr1) [reply phase1/pallavi/main.py:16]: should be just /data

Comment (lawbringr1) [reply phase1/pallavi/main.py:16]: get data folder in current working directory

File: phase1/pallavi/main.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            columns = infer_table_schema(sample_chunk)
+
+            create_table(engine, table_name, columns)
+            load_file_to_db(engine, file_path, table_name)
+            
+            logging.info(f'Processed file {index + 1}/{len(files)}: {file_path.name}')
+
+        logging.info(f"Finished loading {len(files)} files into the MySQL database '{database}'.")
+
+    except Exception as e:
```
Comment (lawbringr1) [inline phase1/pallavi/main.py:38]: what is e?

File: phase1/pallavi/file_processing.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        except SQLAlchemyError as e:
+            logging.error(f"Error inserting data: {e}")
+            raise
+
+def check_directory_structure(directory_path):
+    if not directory_path.is_dir():
+        raise NotADirectoryError(f"{directory_path} is not a directory.")
+    if not any(directory_path.iterdir()):
+        raise FileNotFoundError(f"Directory '{directory_path}' is empty.")
+try:
```
Comment (lawbringr1) [inline phase1/pallavi/file_processing.py:48]: break this method you are doing everything in check_directory_structure add creating db

File: phase1/pallavi/main.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+from file_processing import check_directory_structure
+
+def main():
+        database = 'data'
+        user = '****'
+        password = quote('******')
+        host = '********'
+        cwd = Path(os.getcwd())
+        directory_path = cwd / 'data'
+        engine = get_engine(user, password, host, database)
```
Comment (lawbringr1) [inline phase1/pallavi/main.py:17]: where are you calling your functions?

Comment (lawbringr1) [inline phase1/pallavi/main.py:17]: how do we know this is main method?

**Review by lawbringr1 — APPROVED** [review]

**Pull request merged**